### DISCUSSION TITLE
Cleaning the git history of a large repo

### DISCUSSION THREAD

**Discussion by Bisaloo** [epiverse-trace/epiverse-trace.github.io#311] [Blog post ideas]
This was discussed in the past with @ntorresd based on our experience with serofoi. It was later done by @sbfnk in EpiNow2 as well.

I think a blog post would be very helpful to:

- explain the issues with having a large repo
- explain the causes of a large repo, even if it doesn't seem to include large files in the current `HEAD` version
- present the recommended workflow:
  - make a backup
  - understand where the size issue comes from. Which files in history should be removed?
  - remove the files
  - push the new history
  - communicate the change and invite contributors to clone again
- present the gotchas:
  - can disrupt contributor flow 
  - may or may not automatically close open PR. I'm not entirely clear on this one. I remember we had this issue with serofoi but it didn't seem to happen for EpiNow2 so maybe we did something different, or GitHub has changed something from their end.
  - ...

**Comment by chartgerink** [comment]
Coincidentally today, I found an application that aims to make cleaning the history of a repo more intuitive. I haven't tried it out, but might be of interest in this: https://retcon.app

**Reply by Bisaloo** [reply]
Thanks, this looks great!

I have edited the title to make it a little bit more specific: the intent here is potentially to talk about fixing git repos that became very large by rewriting history. I would be worth checking if this tool can do this kind of this because that's quite tedious at this time.
